### PR TITLE
Added getGuildChannelById methods to ShardManager

### PR DESCRIPTION
**Pull request by Artuto**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

title

### CODE REVIEW

**Comment by Andre601** [comment]
I'm wondering, where you would use a GuildChannel over a Text or PrivateChannel, which already have stuff declared about what you can or cannot do.

**Comment by MinnDevelopment** [comment]
I would prefer if we would just call the method on the individual JDA instances for easier maintanance

**Comment by Artuto** [comment]
Done

File: src/main/java/net/dv8tion/jda/api/sharding/ShardManager.java [hunk 1]
```diff
@@ -795,6 +795,131 @@ default Role getRoleById(@Nonnull final String id)
...
+     *
+     * @param  type
+     *         The {@link net.dv8tion.jda.api.entities.ChannelType}
+     * @param  id
+     *         The ID of the channel
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided ID is null
+     * @throws java.lang.NumberFormatException
+     *         If the provided ID is not a snowflake
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/sharding/ShardManager.java]: What if the type is null?

File: src/main/java/net/dv8tion/jda/api/sharding/ShardManager.java [hunk 2]
```diff
@@ -795,6 +795,131 @@ default Role getRoleById(@Nonnull final String id)
...
+     *     <li>{@link #getStoreChannelById(long)}</li>
+     *     <li>{@link #getCategoryById(long)}</li>
+     * </ul>
+     *
+     * @param  type
+     *         The {@link net.dv8tion.jda.api.entities.ChannelType}
+     * @param  id
+     *         The ID of the channel
+     *
+     * @return The GuildChannel or null
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/sharding/ShardManager.java]: No throws docs for `notNull(type)`

Comment (Artuto) [reply src/main/java/net/dv8tion/jda/api/sharding/ShardManager.java]: I copied the docs from the original methods https://github.com/DV8FromTheWorld/JDA/blob/master/src/main/java/net/dv8tion/jda/api/JDA.java#966-1016

Do you want me to add the missing throws there too?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/sharding/ShardManager.java]: Yes, please do.

**Comment by Artuto** [comment]
Done

**Pull request merged**